codex-process-feasibility/src/test/java/de/netzwerk_universitaetsmedizin/codex/processes/feasibility/service/DownloadFeasibilityResourcesTest.java

```diff
@@ -0,0 +1,303 @@
+
+    private Map<String, List<String>> createSearchQueryParts(String measureId) {
+        return Map.of("_id", Collections.singletonList(measureId), "_include",
+                Collections.singletonList("Measure:depends-on"));
+    }
+
+    @Test
+    public void testDoExecute_NoMeasureReference() {
+        when(execution.getVariable(BPMN_EXECUTION_VARIABLE_TASK)).thenReturn(task);
+        when(taskHelper.getFirstInputParameterReferenceValue(task, CODESYSTEM_FEASIBILITY,
+                CODESYSTEM_FEASIBILITY_VALUE_MEASURE_REFERENCE))
+                .thenReturn(Optional.empty());
+
+        assertThrows(RuntimeException.class, () -> service.execute(execution));
+        verify(task).setStatus(Task.TaskStatus.FAILED);
+    }
+
+    @Test
+    public void testDoExecute_BundleWithTooFewResultEntriesFromLocal() {
+        final String measureId = "id-151003";
+        final Reference measureRef = new Reference("/Measure/" + measureId);
+
+        when(execution.getVariable(BPMN_EXECUTION_VARIABLE_TASK))
+                .thenReturn(task);
+        when(taskHelper.getFirstInputParameterReferenceValue(task, CODESYSTEM_FEASIBILITY,
+                CODESYSTEM_FEASIBILITY_VALUE_MEASURE_REFERENCE))
+                .thenReturn(Optional.of(measureRef));
+        when(clientProvider.getLocalWebserviceClient())
+                .thenReturn(webserviceClient);
+        when(webserviceClient.searchWithStrictHandling(ArgumentMatchers.<Class<MeasureReport>>any(), eq(createSearchQueryParts(measureId))))
+                .thenReturn(new Bundle());
+        when(webserviceClient.getBaseUrl()).thenReturn("http://localhost");
+
+        assertThrows(RuntimeException.class, () -> service.execute(execution));
+        verify(task).setStatus(Task.TaskStatus.FAILED);
+    }
+
+    @Test
+    public void testDoExecute_BundleWithTooFewResultEntriesFromRemote() {
+        final String measureId = "id-151003";
+        final Reference measureRef = new Reference("http://remote.host/Measure/" + measureId);
+
+        when(execution.getVariable(BPMN_EXECUTION_VARIABLE_TASK))
+                .thenReturn(task);
+        when(taskHelper.getFirstInputParameterReferenceValue(task, CODESYSTEM_FEASIBILITY,
+                CODESYSTEM_FEASIBILITY_VALUE_MEASURE_REFERENCE))
+                .thenReturn(Optional.of(measureRef));
+        when(clientProvider.getRemoteWebserviceClient("http://remote.host"))
+                .thenReturn(webserviceClient);
+        when(webserviceClient.searchWithStrictHandling(ArgumentMatchers.<Class<MeasureReport>>any(), eq(createSearchQueryParts(measureId))))
+                .thenReturn(new Bundle());
+        when(webserviceClient.getBaseUrl()).thenReturn("http://remote.host");
+
+        assertThrows(RuntimeException.class, () -> service.execute(execution));
+        verify(task).setStatus(Task.TaskStatus.FAILED);
+    }
+
+    @Test
+    public void testDoExecute_FirstBundleEntryIsNoMeasureFromLocal() {
```

Comment:
It is sufficient to test the local branch once. You did this already above. All other branches should be remote, because the feasibility reasources are always downloaded from remote.